components/StakingCard/StakingCard.tsx

```diff
@@ -0,0 +1,67 @@
+import Image from "next/image"
+import { useState } from "react"
+import { toast } from "react-toastify"
+import { allChains, useNetwork, useSigner, useSwitchNetwork } from "wagmi"
+import getIpfsLink from "../../lib/getIpfsLink"
+import { processStaking } from "../../lib/staking"
+
+const StakingCard = ({ token, stakedTokens, onSuccess, nftContract }) => {
+  const myTokenId = parseInt(token.id.tokenId, 16)
+  const imgHash = token.metadata?.image
+  const isInvalid = imgHash?.includes?.("imgUri") || imgHash?.includes?.("Hello World")
+  const imageUrl = isInvalid ? "" : getIpfsLink(imgHash)
+  const isStaked = stakedTokens.includes(myTokenId.toString())
+  const { data: signer } = useSigner()
+  const { chain } = useNetwork()
+  const { switchNetwork } = useSwitchNetwork()
+  const [isProcessing, setIsProcessing] = useState<boolean>(false)
+  const handleClick = async () => {
+    if (!signer) {
+      toast.error("please connect wallet")
+      return
+    }
+    if (chain.id !== Number(process.env.NEXT_PUBLIC_CHAIN_ID)) {
+      await switchNetwork(Number(process.env.NEXT_PUBLIC_CHAIN_ID))
+      const myChain = allChains.find(
+        (blockchain) => blockchain.id === parseInt(process.env.NEXT_PUBLIC_CHAIN_ID, 10),
+      )
+      toast.error(`Please connect to ${myChain.name} and try again`)
+      return
+    }
+    if (isStaked) {
+      await processStaking(nftContract, myTokenId, onSuccess, false, setIsProcessing)
+    } else {
+      await processStaking(nftContract, myTokenId, onSuccess, true, setIsProcessing)
```

Comment:
can you just pass in the 4th param as a variable instead of making an if function for 2 identical function calls?

Reply:
```suggestion
      await processStaking(nftContract, myTokenId, onSuccess, !isStaked, setIsProcessing)
```

this suggestion is missing a deletion on line 35 but should reduce code and be easier to read.